.circleci/config.yml

```diff
@@ -42,7 +47,53 @@ jobs:
                 paths:
                     - ~/.cargo
                     - target
+  win-build:
+    description: |
+      Build the crate for windows.
+    executor:
+      name: win/default
+      shell: powershell.exe
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - cargo-{{ checksum "Cargo.lock" }}-v1
+      - run:
+          name: Install rustup
+          command: choco install rustup.install
+      - run:
+          name: Install clang
+          command: choco install llvm
+      - run:
+          name: Add target
+          command: rustup target add x86_64-pc-windows-msvc
+      - run:
+          name: Install target toolchain
+          command: rustup toolchain install stable-x86_64-pc-windows-msvc
+      # Run the commands without using the orbs since Clippy needs a little longer then 10 minutes
+      - run:
+          name: Install Clippy
+          command: rustup component add clippy
+      - run:
+          name: Run Clippy
+          # Remove the first two lines of gitconfig as work around
```

Comment:
As a workaround for what?

Reply:
https://github.com/rust-lang/cargo/issues/2078
https://github.com/rust-lang/cargo/issues/2422
https://app.circleci.com/pipelines/github/carver/trin/147/workflows/d059c7c6-d759-4f79-a96c-99c16ba078ab/jobs/177

Here is some context on it.
 
```
Updating crates.io index
error: failed to get `base64` as a dependency of package `trin v0.1.0 (C:\Users\circleci\project)`

Caused by:
  failed to load source for dependency `base64`

Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to authenticate when downloading repository: ssh://git@github.com/rust-lang/crates.io-index

  * attempted ssh-agent authentication, but no usernames succeeded: `git`

  if the git CLI succeeds then `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  error authenticating: failed connecting agent; class=Ssh (23)
```

So that was the "fix" I could find from googling it.